Console release notes 2022-12-01 (#357)
neondatabase/website: content/release-notes/2022-12-01.md

---
label: 'Console'
---

### What's new

> - Control plane: List operations in V2 API supports cursor pagination.

Reviewer: ```suggestion
- API: List operations in V2 API supports cursor pagination.
```

Reviewer 2: fixed.

> - Control plane: Added validation for empty names for branches and 

Reviewer: ```suggestion
- API: Added validation for empty names for branches and 
```

Reviewer 2: fixed.

projects. 

> - Control plane: Added LSN validation.

Reviewer: ```suggestion
- API: Added LSN validation.
```

> - Console: Added 'delete endpoint' button.

Reviewer: ```suggestion
- UI: Added 'delete endpoint' button.
```

> - Control plane: Make parent_id and name optional when creating a branch.

Reviewer: ```suggestion
- API: Make parent_id and name optional when creating a branch.
```

> - Console: Show branch_id and fork button in branch view.

Reviewer: ```suggestion
- UI: Show branch_id and fork button in branch view.
```

- Control plane: Added new autoscaling limits.
- Control plane: Added region maintenance/testing flag.

> - Control plane: Return complete host for connecting in endpoint response.

Reviewer: ```suggestion
- API: Return complete host for connecting in endpoint response.
```

> - Control plane: Rename default database from `main` to `neondb`.

Reviewer: ```suggestion
- Control plane: Rename the database that is created along with the project from `main` to `neondb`.
```

> - Control plane: Add per-project branch/endpoint limits.

Reviewer: ```suggestion
- UI: Add per-project branch/endpoint limits.
```

not sure about this one, maybe we should call out the update of the free tier here instead.

> - Control plane: Support allow_connections flag for an endpoint.

Reviewer: ```suggestion
- API: Support allow_connections flag for an endpoint.
```

> - Console: Allow to choose an endpoint in passwordless auth.

Reviewer: ```suggestion
- UI: Allow to choose an endpoint in passwordless auth.
```

- Control plane: Added project maintenance flag.

> - Control plane: Allow to change the branch of an endpoint.

Reviewer: ```suggestion
- API: Allow to change the branch of an endpoint.
```

> - Control plane: Allow creation of a branch without an endpoint.

Reviewer: ```suggestion
- API: Allow creation of a branch without an endpoint.
```

> - Control plane: Creating a branch with V1 is now deprecated.

Reviewer: ```suggestion
- API: Creating a branch with V1 is now deprecated.
```

> - Console: Added API response examples.

Reviewer: ```suggestion
- API: Added response examples to the API Spec.
```

### Bug fixes

- UI: Fixed rerender during endpoint polling.
- UI: Stop polling on project deletion.

> - Control Plane: Return correct error on duplicate branch name.

Reviewer: ```suggestion
- API: Return correct error on duplicate branch name.
```

- UI: Handle errors when polling endpoint status.
- Control plane: Handle ID collisions when creating a branch or endpoint.

> - Control plane: Handle repeated endpoint suspension.

Reviewer: ```suggestion
- API: Handle repeated endpoint suspension.
```

not sure about this one

- UI: don't show internal service roles.

> - Control plane: fix ordering of operations and projects.

Reviewer: ```suggestion
- API: fix ordering of operations and projects.
```

> - UI: Handle sign-in during passwordless authentication.

Reviewer: ```suggestion
- UI: Handle sign-in during passwordless connect.
```
